### PR TITLE
[selenium-webdriver] Added util library

### PR DESCRIPTION
**Pull request by TamsilAmani**
Please fill in this template.

Motivation : https://github.com/SeleniumHQ/selenium/issues/11184

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/selenium-webdriver/lib/util.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Determines whether a {@code value} should be treated as an object.
+ * @param {?} value The value to test.
+ * @returns {boolean} Whether the value is an object.
+ */
+export function isObject(value: any): boolean;
```
Comment (SupernaviX) [inline types/selenium-webdriver/lib/util.d.ts:6]:
```suggestion
export function isObject(value: any): value is object;
```

This function looks like a [type guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates), so might as well make it one

File: types/selenium-webdriver/lib/util.d.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * Determines whether a {@code value} should be treated as a promise.
+ * Any object whose "then" property is a function will be considered a promise.
+ *
+ * @param {?} value The value to test.
+ * @return {boolean} Whether the value is a promise.
+ */
+export function isPromise(value: any): boolean;
```
Comment (SupernaviX) [inline types/selenium-webdriver/lib/util.d.ts:15]:
```suggestion
export function isPromise(value: any): value is Promise<any>;
```
This function looks like a [type guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates), so might as well make it one

**Review by SupernaviX — COMMENTED** [review]
This looks OK, but I think that these should be type guards.

Also, I don't see these functions in the [API docs](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver.html). They look like they might be internal, and not meant to be exposed to consumers.

File: types/selenium-webdriver/lib/util.d.ts (hunk 1 above)
Comment (TamsilAmani) [reply types/selenium-webdriver/lib/util.d.ts]: Changes incorporated

File: types/selenium-webdriver/lib/util.d.ts (hunk 2 above)
Comment (TamsilAmani) [reply types/selenium-webdriver/lib/util.d.ts]: Changes incorporated

**Comment by TamsilAmani** [comment]
@SupernaviX Thank you for the review suggestions. You help me a lot with TS. I have made the suggested changes and I think util should be there to keep it in line with JS binding.

**Review by SupernaviX — APPROVED** [review]
While I still think it's strange to expose this internal-looking "util" in typings, I'm not actually a selenium maintainer and the typings themselves look right to me. So :shipit:

**Pull request merged**